api/v1alpha1/flinkcluster_validate.go

```diff
@@ -269,3 +279,23 @@ func (v *Validator) validateCleanupAction(
 	}
 	return nil
 }
+
+func (v *Validator) validateMemoryOffHeapRatio(
+	offHeapRatio *int32, component string) error {
+	if offHeapRatio == nil || *offHeapRatio > 100 || *offHeapRatio < 0 {
+		return fmt.Errorf("invalid %v memoryOffHeapRatio, it must be between 0 and 100", component)
+	}
+	return nil
+}
+
+func (v *Validator) validateMemoryOffHeapMin(
+	offHeapMin *int32, memoryLimit *resource.Quantity, component string) error {
+	divisor := resource.MustParse("1Mi")
+	memLimit := math.Floor(float64(memoryLimit.Value()) / float64(divisor.Value()))
+	if offHeapMin == nil {
+		return fmt.Errorf("invalid %v memory configuration, MemoryOffHeapMin is not specified", component)
+	} else if memLimit > 0 && *offHeapMin > int32(memLimit) {
+		return fmt.Errorf("invalid %v memory configuration, memory limit must be larger than MemoryOffHeapMin", component)
```

Comment:
Can we include the actual values in the error message?

Reply:
I have included them in last commit.